alacritty_terminal/src/grid/row.rs

```diff
@@ -80,11 +84,17 @@ impl<T: Copy> Row<T> {
     }
 
     /// Resets contents to the contents of `other`
-    pub fn reset(&mut self, other: &T) {
-        for item in &mut self.inner[..] {
-            *item = *other;
+    pub fn reset(&mut self, template: &T)
+    where
+        T: GridCell,
+    {
+        for item in &mut self.inner[..self.occ] {
+            *item = *template;
+        }
+
+        if template.is_empty() {
+            self.occ = 0;
```

Comment:
Sorry, I think I'm clearly not understanding the template quite right. But I'd expect the `self.occ` to be set to the template's length. Or something like that here.

Reply:
The template is a single cell. So we're resetting every occupied cell with the template cell.

Now if the cell we're using to replace it is empty, we know that none of the cells are occupied anymore and we can set it to 0.

However if the cell is not empty, we know that all cells in a row might have changed, however they are all still occupied.

Originally I iterated over each cell and checked if it's empty, setting the `occ` to the last element which is not empty. But since all elements are the same (they are the template cell), this is an optimized way of doing the same.

Reply:
Ahhhhh, I understand. Setting all cells to an empty template collapses the whole thing to 0, otherwise it stays the same.

A template is always essentially length 1 or 0.

Reply:
That makes sense, thanks for clearing that up for me.

Reply:
Maybe add a comment is a good idea, since this is not something obvious?